orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -689,21 +689,17 @@ def get_providers(provider_options):
                     providers[providers.index(provider_name)] = (provider_name, provider_options[provider_name])
             #default: using cuda
             elif 'cuda' in self.options.device.id.lower():
-                cuda_ep_options = {"device_id": _utils.get_device_index(self.options.device.id)}
-
-                cuda_ep_name = ("ROCMExecutionProvider" if self.is_rocm_pytorch else "CUDAExecutionProvider")
+                gpu_ep_options = {"device_id": _utils.get_device_index(self.options.device.id)}
+                gpu_ep_name = ("ROCMExecutionProvider" if self.is_rocm_pytorch else "CUDAExecutionProvider")
                 if self.options.device.mem_limit > 0:
-                    if not self.is_rocm_pytorch:
-                        cuda_ep_options["cuda_mem_limit"] = self.options.device.mem_limit
-                    else:
-                        warnings.warn("Ignoring 'mem_limit' for {}".format(cuda_ep_name))
+                    gpu_ep_options["gpu_mem_limit"] = self.options.device.mem_limit
```

Comment:
Looks like a good change - any concern it may break existing ORTTrainer-based models?

Reply:
Yeah, it could break existing ORTTrainer-based models if this option is used. But, this maybe a good time to do this change since we are moving to ORTModule.

Reply:
Besides, we have this warning for ORTTrainer. 

/opt/conda/lib/python3.6/site-packages/onnxruntime/capi/ort_trainer.py:628: UserWarning: DISCLAIMER: This is an early version of an experimental training API and it is subject to change. DO NOT create production applications with it
  warnings.warn('DISCLAIMER: This is an early version of an experimental training API and it is subject to change. DO NOT create production applications with it')